File: Api/MySerialList.Model/FilmProduction/FilmProductionRating.cs
﻿using System;

namespace MySerialList.Model.FilmProduction
{
    public class FilmProductionRating
    {
        public string Title { get; set; }
        public string Genre { get; set; }
        public string Released { get; set; }
        public int FilmProductionId { get; set; }
        public string Poster { get; set; }
        public string Plot { get; set; }
        public int Votes { get; set; }
        public double Rating { get; set; }
        public int? MyRating { get; set; }
        public bool IsSeries { get; set; }
        public int? Seasons { get; set; }
        public int TotalEpisodes { get; set; }
        public int WatchedEpisodes { get; set; }
        public bool CurrentUserItem { get; set; }
        public bool Last { get; set; }
    }
}

File: my_serial_list/lib/features/film_production/domain/entities/film_production_rating.dart
import 'package:equatable/equatable.dart';
import 'package:meta/meta.dart';

class FilmProductionRating extends Equatable {
  final String title;
  final String genre;
  final String released;
  final int filmProductionId;
  final String poster;
  final String plot;
  final int votes;
  final double rating;
  final bool isSeries;
  final int seasons;
  final bool last;
  final bool currentUserItem;
  final int totalEpisodes;

  FilmProductionRating({
    @required this.title,
    @required this.genre,
    @required this.released,
    @required this.filmProductionId,
    @required this.poster,
    @required this.plot,
    @required this.votes,
    @required this.rating,
    @required this.isSeries,
    @required this.seasons,
    @required this.last,
    @required this.currentUserItem,
    @required this.totalEpisodes,
  });

  @override
  List<Object> get props => [
        title,
        genre,
        released,
        filmProductionId,
        poster,
        plot,
        votes,
        rating,
        isSeries,
        seasons,
        last,
        currentUserItem,
        totalEpisodes,
      ];
}
